src/v/raft/vote_stm.cc

```diff
@@ -126,71 +126,74 @@ ss::future<election_success> vote_stm::vote(bool leadership_transfer) {
+        return _ptr->_op_lock
+          .with([this, leadership_transfer] {
+              _config = _ptr->config();
+              // check again while under op_sem
+              if (_ptr->should_skip_vote(leadership_transfer)) {
+                  return ss::make_ready_future<prepare_election_result>(
+                    prepare_election_result::skip_election);
+              }
+              // 5.2.1 mark node as candidate, and update leader id
+              _ptr->_vstate = consensus::vote_state::candidate;
+              //  only trigger notification when we had a leader previously
+              if (_ptr->_leader_id) {
+                  _ptr->_leader_id = std::nullopt;
+                  _ptr->trigger_leadership_notification();
+              }
+
+              if (_prevote && leadership_transfer) {
+                  return ssx::now(prepare_election_result::immediate_success);
+              }
+
+              // 5.2.1.2
+              /**
+               * Pre-voting doesn't increase the term
+               */
+              if (!_prevote) {
+                  _ptr->_term += model::term_id(1);
+                  _ptr->_voted_for = {};
+              }
+
+              // special case, it may happen that node requesting votes is not a
+              // voter, it may happen if it is a learner in previous
+              // configuration
+              _replies.emplace(_ptr->_self, *this);
+
+              // vote is the only method under _op_sem
+              _config->for_each_voter(
+                [this](vnode id) { _replies.emplace(id, *this); });
+
+              auto lstats = _ptr->_log->offsets();
+              auto last_entry_term = _ptr->get_last_entry_term(lstats);
+
+              _req = vote_request{
+                .node_id = _ptr->_self,
+                .group = _ptr->group(),
+                .term = _ptr->term(),
+                .prev_log_index = lstats.dirty_offset,
+                .prev_log_term = last_entry_term,
+                .leadership_transfer = leadership_transfer};
+              // we have to self vote before dispatching vote request to
+              // other nodes, this vote has to be done under op semaphore as
+              // it changes voted_for state
+              return self_vote().then(
+                [] { return prepare_election_result::proceed_with_election; });
+          })
+          .then([this](prepare_election_result result) {
+              switch (result) {
+              case prepare_election_result::skip_election:
+                  return ss::make_ready_future<election_success>(
+                    election_success::no);
+              case prepare_election_result::proceed_with_election:
+                  return do_vote();
+              case prepare_election_result::immediate_success:
+                  return ss::make_ready_future<election_success>(
+                    election_success::yes);
+              }
+          });
+    });
```

Comment:
nit: can you separate this into two commits where in first one you add a lock and in second one apply code formatting ?

Reply:
It's not going to be straightforward, as my IDE (vscode) applies clang-format on save. If it's part of our contribution rules or traditions could you tell me how you normally do it?

Both [github](https://github.com/redpanda-data/redpanda/pull/21361/commits/2223844693cbe01fe4dcfb39258ba03e40d24f32?diff=unified&w=1#diff-a3db863b3d7487238845de3844ea835d2fe43c0cebcd4d30fa931520aac53931) and `git diff` support word diff. Line-wise difference is not an intrinsic property of the change, but rather way you present it.

I can do it if you think it should be done, but I'm curious about the reasons.

Reply:
good point, it was a tradition, i wasn't aware that this is the case